src/content/docs/en/guides/assets.mdx

```diff
@@ -0,0 +1,213 @@
+
+- Brand new, more intuitive `Image` component
+- Automatic image optimization in Markdown and MDX files
+- Integration with [Content Collections](/en/guides/content-collections/)
+- Improved error messages and types
+
+:::caution
+Assets are an _experimental_ Astro feature. Be mindful that the API is subject to change before it is marked as stable.
+:::
+
+## **Enabling Assets in Your Project**
+
+To enable built-in asset support, you must add the following lines to your `astro.config.mjs` configuration file.
+
+```diff
+ // astro.config.mjs
+ import { defineConfig } from 'astro/config';
+
+ export default defineConfig({
++ experimental: {
++   assets: true
++ }
+ });
+```
+
+## New signature for images imports
+
+Previously, importing an image would return a simple `string` with the path of the image. With the new `image` features enabled, imported image assets now match the following signature:
+
+```ts
+interface ImageAsset {
+  src: string;
+  width: number;
+  height: number;
+  format: string;
+}
+```
+
+## New `src/assets` directory
+
+We recommend putting images in the `src/assets` directory. This is entirely optional, but will become the recommended location for other asset types in the future.
+
+Your images stored in `src/assets/` can be used by components (`.astro`, `.mdx` and other UI frameworks) and in Markdown files.
+
+Import them from a **relative file path** or [import alias](/en/guides/aliases/) in any component file and then use the import as the `Image` component's `src` attribute.
+
+```astro
+---
+// src/pages/index.astro
+import { Image } from 'astro:assets';
+
+// Access images in `src/images/`
+import logo from '../assets/images/logo.png';
+---
+<Image src={logo} alt="Astro" />
+```
+
+## New Markdown support
+
+Relative images can now be referenced in Markdown and MDX files. These will automatically be optimized and hashed by Astro's build process.
+
+Remote images are also valid, but are not automatically optimized by Astro's build process.
+
+```md
+<!-- src/pages/post-1.md -->
+# My Markdown Page
+
+<!-- Local image stored at src/assets/stars.png -->
+![A starry night sky.](../../assets/stars.png)
+<img src="/_assets/stars.hash.png" alt="A starry night sky.">
+
+<!-- Remote image on another server -->
+![Astro](https://astro.build/assets/logo.png)
+<img src="https://astro.build/assets/logo.png" width="25" alt="Astro">
+```
+
+### Content Collections
+
+A common use case for blog posts is to reference assets in their metadata (think, cover image for an article, picture for an author etc).
+
+With the `src/assets` folder, we're also introducing a new `asset` helper to the `astro:content` module. The `asset` helper will validate that the user references a valid asset from the `src/assets` folder:
+
+```ts
+import { asset, defineCollection, z } from "astro:content";
+
+const blogCollection = defineCollection({
+  schema: z.object({
+    title: z.string(),
+    image: asset({ width: 1120, height: 1120 }),
```

Comment:
From what I understand, this isn't _validating_ that the asset has width 1120 and height 1120, but _transforming_ it, similar to Zod's `.transform`. We can make this more clear.

Reply:
This is validating it. If your image doesn't have the requested width / height / format, you'll get an error

Reply:
What it means is that the final shape you get from using getCollectionEntry later for the image is the same as if you've had imported it.

`asset` does transform in that regard (you pass a path, you get an object with different properties, like when you import an image), but it does not use the parameters in your schema to resize or optimize the image, just to validate the source file